lib/setup/profile.js

```diff
@@ -77,33 +177,53 @@ var appendScript = function(name, options) {
 module.exports.update = function() {
   var handled = false;
   var changed = false;
+  var appendFile;
 
-  var promise = Promise.map(['.bash_profile', '.zshrc'], function(name) {
-    return appendScript(name);
-  })
-  .each(function(result) {
-    if (result) {
-      handled = true;
-      changed = changed || (result === 'change');
+  var promise = shellInitScriptChooser()
+  .then(function(initScript) {
+    appendFile = initScript;
+
+    if (appendFile === bashProfileFileName) {
```

Comment:
What's the purpose of this `if`/`else` here?

Why can't it just update both `.bash_profile` and `.zshrc` _or_ `.bashrc` and `.zshrc`? I know it's uncommon, but some people will use multiple shells, so if they have both config files, why not support them both?